### DISCUSSION TITLE
Tuya Smart vs Smart Life

### DISCUSSION THREAD

**Discussion by Shwalamazula** [make-all/tuya-local#1562] [Q&A] [answered]
I am having a hard time getting my Tuya devices to connect to Home Assistant. I can do it thru the cloud integration, but want to use local control. I have 3 devices (2 lamps and a heat pad). I have tried using LocalTuya and just recently tried with Tuya Local. Neither will connect to my devices. I have local keys and can ping the devices just fine from the Tuya cloud API.

I have tried a bunch of troubleshooting including:

- Deleting and re-adding devices to app
- Creating new project in cloud
- Setting up static IP
- Forcing a device reconnect
- Closing app
- Powering down phone (in case the devices are looking for my phone)

I was about to try TinyTuya when I saw that it was specifically stated that the devices had to be connected to the Smart Life app. Is it possible that there is some configuration difference between the devices connected to Smart Life vs Tuya Smart? HA sees the devices as Tuya and tries to launch the cloud integration, so I know the devices must be visible.

**Comment by make-all** [comment]
If you are able to use your devices with the Tuya cloud integration, you will be able to use them with this integration and tinytuya directly.  Smart Life and Tuya Smart both work, the important thing is that they are pairing your devices with the default Tuya cloud, as many manufacturers own branded apps pair the devices with a partitioned off portion of the Tuya Cloud that is only for that manufacturer, so you cannot access them through the developer APIs.  It sounds like you are having local connection issues, for which there might be some steps in #522 that can help you troubleshoot.

**Reply by Shwalamazula** [reply]
https://github.com/make-all/tuya-local/discussions/522 was where I started this fun journey. I have tried all of the solutions suggested in that thread other than trying to connect with tinytuya.

Is there a debugging log I can turn on for Tuya Local? I looked at my core log and all I have is this (even though I have done multiple connection attempts):

2024-01-24 09:40:57.097 ERROR (MainThread) [custom_components.tuya_local.device] <class 'Exception'>: Unable to find device on network (specify IP address) while initialising device eb472320fed523bc10aycg

I don't know if it makes any difference but I am running a Home Assistant Yellow strictly over LAN. All wireless coming into HA goes thru my Wi-Fi router.

**Reply by make-all** [reply]
It may mean that your networks are completely separate and HA is not able to connect to anything on your WiFi network.

Do you have shell access to the HA box so you can try pinging the WiFi devices?

**Reply by Shwalamazula** [reply, marked as answer]
I figured it out. Per your suggestion, I tried to ping the devices from HA terminal and got no response. I also tried to ping from my PC and got no response (even though I was logged into my router and could clearly see that the devices were connected. Out of curiosity, I looked at my IoT guest network settings and Intranet Access was disabled (which is what I would expect from a guest network). As an experiment, I enabled Intranet Access. As soon as the devices all re-connected, I was able to ping them and add them to HA.

I am guessing the reason my other devices on that guest network are not having a problem is because they are zigbee and they make their own mesh network. Now I need to do some thinking about how I want to partition my network or if I just need to throw everything on the same one and just set up strict access rules for my PC. 

Thanks for the assist.

**Reply by make-all** [reply]
I don't know how flexible your router is, but ideally you would be able to configure your guest network so that it had access to Home Assistant and nothing else on the intranet.  Or vice versa, put your Home Assistant on the guest network and configure it so that only Home Assistant has access from the guest network to the intranet (perhaps only for connections established from the intranet if you only need web access to Home Assistant).  But these are probably quite difficult to achieve with most consumer grade routers, as they simplify the firewalling options to make complex rules impossible to set up.

**Reply by Shwalamazula** [reply]
I'll look into that.
